### PR TITLE
Integration Test for AttributeGen WASM Filter

### PR DESCRIPTION
**Pull request by vicentefb**
This PR includes the following: 

- Integration Test for AttributeGen WASM filter. 
- Go code, YAML and template file with plugin configuration.

### CODE REVIEW

File: testdata/metric/server_request_total.yaml.tmpl [hunk 1]
```diff
@@ -47,6 +47,9 @@ metric:
     value: http
     {{- end }}
   - name: response_code
+    {{- if .Vars.ResponseCodeClass }}
+    value: {{ .Vars.ResponseCodeClass }}
+    {{- else }}
     value: "200"
   - name: grpc_response_status
```
Comment (vicentefb) [inline testdata/metric/server_request_total.yaml.tmpl:54]: Forgot an {{- end }} statement here.

**Comment by kyessenov** [comment]
Hi, do you mind adding a description to the PR? Is this a draft?

**Comment by bianpengyuan** [comment]
@kyessenov Yeah this is a draft to add integration test for attribute gen. I am helping on debugging the issue.

@vicentefb you don't necessarily need to make a PR for the draft change. Pointing me to your branch on github is good enough.

**Comment by bianpengyuan** [comment]
Can you update title and comment?

File: test/envoye2e/stats_plugin/stats_test.go [hunk 2]
```diff
@@ -261,7 +275,7 @@ func TestStatsGrpc(t *testing.T) {
 	params.Vars["ServerMetadata"] = params.LoadTestData("testdata/server_node_metadata.json.tmpl")
 	params.Vars["ServerHTTPFilters"] = params.LoadTestData("testdata/filters/stats_inbound.yaml.tmpl")
 	params.Vars["ClientHTTPFilters"] = params.LoadTestData("testdata/filters/stats_outbound.yaml.tmpl")
-
+	fmt.Println("THIS IS SERVERHTTPFILTERS", params.Vars["ClientHTTPFilters"])
```
Comment (bianpengyuan) [inline test/envoye2e/stats_plugin/stats_test.go:278]: Needs some clean up. Looks like lots of place are not correctly formatted and debug log left

**Comment by bianpengyuan** [comment]
@richardwxn fyi.

File: test/envoye2e/stats_plugin/stats_test.go (hunk 2 above)
Comment (vicentefb) [reply test/envoye2e/stats_plugin/stats_test.go:278]: okok on it

**Comment by bianpengyuan** [comment]
Can you update the PR? It is failing lint check and having merge conflict.

File: test/envoye2e/stats_plugin/stats_test.go [hunk 3]
```diff
@@ -123,6 +123,20 @@ var TestCases = []struct {
 	},
 }
 
+var AttributeGenRuntimes = []struct {
+	AttributeGenFilterCode  string
+	WasmRuntime             string
+}{
+	{
+		AttributeGenFilterCode:    "inline_string: \"envoy.wasm.attributegen\"",
+		WasmRuntime:        	     "envoy.wasm.runtime.null",
```
Comment (bianpengyuan) [inline test/envoye2e/stats_plugin/stats_test.go:132]: nit: seems like lots of places in the file that are not properly aligned. For example L132, L136, L324, etc. Could you use gofmt or the tools in your IDE to format the file?

File: test/envoye2e/stats_plugin/stats_test.go [hunk 4]
```diff
@@ -296,3 +309,50 @@ func TestStatsGrpc(t *testing.T) {
...
+				"WasmRuntime":				        "envoy.wasm.runtime.null",
+				"StatsConfig":                driver.LoadTestData("testdata/bootstrap/stats.yaml.tmpl"),
+				"StatsFilterClientConfig":    driver.LoadTestJSON("testdata/stats/client_config.yaml"),
+				"StatsFilterServerConfig":    driver.LoadTestJSON("testdata/stats/stats_filter_config.yaml"),
+				"ResponseCodeClass":		      "2xx",
+			}, envoye2e.ProxyE2ETests)
+			params.Vars["ClientMetadata"] = params.LoadTestData("testdata/client_node_metadata.json.tmpl")
+			params.Vars["ServerMetadata"] = params.LoadTestData("testdata/server_node_metadata.json.tmpl")
+			params.Vars["ServerHTTPFilters"] = params.LoadTestData("testdata/filters/attributegen.yaml.tmpl") 
+			params.Vars["ServerHTTPFilters"] += "\n" + params.LoadTestData("testdata/filters/stats_inbound.yaml.tmpl")
```
Comment (bianpengyuan) [inline test/envoye2e/stats_plugin/stats_test.go:333]:
```suggestion
			params.Vars["ServerHTTPFilters"] = params.LoadTestData("testdata/filters/attributegen.yaml.tmpl") + "\n" +
				params.LoadTestData("testdata/filters/stats_inbound.yaml.tmpl")
```

File: test/envoye2e/README.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
+# Integration Test for WASM Filter
```
Comment (bianpengyuan) [inline test/envoye2e/README.md:1]: This looks great! Though this looks more like a very detailed explanation of the attributegen filter, which will be useful when someone want to understand how to write a test specifically. Could you move it to stats_plugin folder?

**Pull request merged**